src/functionalities/InputNode.js

```diff
@@ -30,7 +30,7 @@ class InputNode extends Functionality {
   static get SCHEMA() {
     return Joi.object({
       source: Joi.object({
-        name: Joi.string().allow(...InputNode.VALID_SOURCE_NAMES).required(),
```

Comment:
we might have messed this up. When a source is provided, the name should be required. Source itself should not be required.

Reply:
good 👁️ !